### ISSUE TITLE
Responsive Breakpoints based on actual table width

### ISSUE THREAD

**Issue by Tekanewa** [DataTables/DataTables#906]
Instead of applying the responsive breakpoints to the available screen width, wouldn't it be more feasible to apply them to the actual table width instead?

Rarely ever will a table actually use the full width of a screen; after internal paddings/margins applied to surrounding containers, the table will almost always have a smaller width than the screen width. The issue compounds even further when the table is embedded inside a column, for example within a bootstrap column grid. In that case, the table width will be significantly less than the actual screen width, causing the breakpoints to be applied too late.

The FooTable script already provides such an option "useParentWidth", which when set to 'true' will use the parent container of the table as basis for its breakpoints, instead of the screen width.

I believe DataTable could benefit from something similar.

**Comment by DataTables** [comment]
> Instead of applying the responsive breakpoints to the available screen width, wouldn't it be more feasible to apply them to the actual table width instead?

Yes. It would be the parent container rather than the table itself (since the table is the element to be resized by the responsive code). I honestly can't remember why I didn't do that initially - it does seem fairly obvious.

This is something that would most certainly be worth looking into.

**Comment by Tekanewa** [comment]
As far as I can tell, the change should be fairly simple. In the "_constructor" function for the "Responsive" prototype, I simply replaced the two references to "$(window).width()" with "$(dt.table().node()).parent().width()", which seems to have done the trick for me. Let me know if this is the wrong approach, but based on my tests, this would use the width of the parent container of the table as basis for the responsive breakpoints.

**Comment by DataTables** [comment]
I think that would be the correct method for automatic hiding - but there is the special case of the [class control](https://datatables.net/extensions/responsive/classes), which will still need the window width. e.g. if a column is set not to display in a mobile display size, we'd need to know that from the window width. So now there is both the container and the window width to consider.
